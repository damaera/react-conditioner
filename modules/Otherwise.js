import React, { PropTypes } from 'react';

import If from './If.js'

const Otherwise = (props) => {
  return (
    <If condition={ true } >
      { props.children }
    </If>
  )
}

Otherwise.propTypes = {
  children: PropTypes.element,
}

export default Otherwise;