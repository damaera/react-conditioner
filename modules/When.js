import React from 'react';

import PropTypes from 'prop-types';

import If from './If.js'

const When = (props) => {
  return <If {...props} />
}

When.propTypes = {
  condition: PropTypes.any,
  children: PropTypes.any,
}

export default When;