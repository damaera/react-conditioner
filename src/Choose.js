import React, { PropTypes } from 'react';

const Choose = (props) => {
  let foundEl = null
  var i = 0

  if (props.children.length) {

    const { children } = props
    for (i = 0; i < children.length; i++) {

      const child = children[i]

      if (child.type.name === 'When') {
        if (child.props.condition) {
          foundEl = <span>{ child }</span>
          break
        }
      } else if (child.type.name === 'Otherwise') {
        if (foundEl === null) {
          foundEl = <span>{ child }</span>
          break
        }
      }

    }
  }
  return foundEl
}

Choose.propTypes = {
}

export default Choose;