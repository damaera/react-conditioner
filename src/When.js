import React from 'react';

import PropTypes from 'prop-types';

import If from './If.js'

const When = (props) => {
  return <If condition={props.condition}>
    { props.children }
  </If>
}

When.propTypes = {
  condition: PropTypes.any.isRequired,
  children: PropTypes.any.isRequired,
}

export default When;