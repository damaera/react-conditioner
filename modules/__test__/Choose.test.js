import React from 'react';
import renderer from 'react-test-renderer';

import Choose from '../Choose'
import When from '../When'
import Otherwise from '../Otherwise'

it('render string when first condition true', () => {
  const tree = renderer.create(
    <Choose>
      <When condition={true}>
        Test 1
      </When>
    </Choose>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('not render second true condition', () => {
  const tree = renderer.create(
    <Choose>
      <When condition={true}>
        Test 1
      </When>
      <When condition={true}>
        Test 2
      </When>
    </Choose>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('render second condition, first is false, and second is true', () => {
  const tree = renderer.create(
    <Choose>
      <When condition={false}>
        Test 1
      </When>
      <When condition={true}>
        Test 2
      </When>
    </Choose>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('render otherwise', () => {
  const tree = renderer.create(
    <Choose>
      <When condition={false}>
        Test 1
      </When>
      <When condition={false}>
        Test 2
      </When>
      <Otherwise>
        Otherwise
      </Otherwise>
    </Choose>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
