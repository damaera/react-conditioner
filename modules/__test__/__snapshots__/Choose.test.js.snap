// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`not render second true condition 1`] = `
<span>
  <span>
    Test 1
  </span>
</span>
`;

exports[`render otherwise 1`] = `
<span>
  <span>
    Otherwise
  </span>
</span>
`;

exports[`render second condition, first is false, and second is true 1`] = `
<span>
  <span>
    Test 2
  </span>
</span>
`;

exports[`render string when first condition true 1`] = `null`;
